sklearn/ensemble/tests/test_forest.py

```diff
@@ -1443,6 +1443,40 @@ def test_max_samples_exceptions(name, max_samples, exc_type, exc_msg):
         est.fit(X, y)
 
 
+@pytest.mark.parametrize('name', FOREST_REGRESSORS)
+def test_max_samples_boundary_regressors(name):
+    X_train, X_test, y_train, y_test = train_test_split(
+        X_reg, y_reg, train_size=0.7, test_size=0.3, random_state=0)
+
+    ms_1_model = FOREST_REGRESSORS[name](max_samples=1.0, random_state=0)
+    ms_1_predict = ms_1_model.fit(X_train, y_train).predict(X_test)
+
+    ms_None_model = FOREST_REGRESSORS[name](max_samples=None, random_state=0)
+    ms_None_predict = ms_None_model.fit(X_train, y_train).predict(X_test)
+
+    ms_1_ms = mean_squared_error(ms_1_predict, y_test)
+    ms_None_ms = mean_squared_error(ms_None_predict, y_test)
+
+    assert ms_1_ms == pytest.approx(ms_None_ms)
+
+
+@pytest.mark.parametrize('name', FOREST_CLASSIFIERS)
+def test_max_samples_boundary_classifiers(name):
+    rng = np.random.RandomState(1)
+
+    X_train = rng.randn(10000, 2)
+    y_train = rng.randn(10000) > 0
+    X_test = rng.randn(1000, 2)
```

Comment:
Can we use `train_test_split` here on `X_large` and `y_large`?

```python
    X_train, X_test, y_train, _ = train_test_split(
        X_large, y_large, random_state=0, stratify=y_large)
```

On my machine, using `X_large` also decreases the runtime of this test from 12s to 1s.

Reply:
fix it on https://github.com/scikit-learn/scikit-learn/pull/20159/commits/7f66d0c601327e76f33a51a8a3d59c3246f39d81